libiocage/lib/helpers.py

```diff
@@ -1,38 +1,31 @@
 import re
 import subprocess
 import uuid
-
-import libzfs
+from typing import Optional
 
 import libiocage.lib.Datasets
 import libiocage.lib.Host
 import libiocage.lib.Logger
+import libzfs
 
 
-def init_zfs(self, zfs):
+def init_zfs(self, zfs: Optional[libzfs.ZFS]) -> libzfs.ZFS:
     if isinstance(zfs, libzfs.ZFS):
-        self.zfs = zfs
+        return zfs
     else:
-        self.zfs = get_zfs()
+        return get_zfs()
 
 
 def get_zfs():
     return libzfs.ZFS(history=True, history_prefix="<iocage>")
 
 
-def init_host(self, host=None):
+def init_host(host: libiocage.lib.Host.Host=None, logger:
+              libiocage.lib.Logger.Logger=None) -> libiocage.lib.Host.Host:
```

Comment:
this return type is now incompatible

Reply:
We're returning a HostGenerator. Internally we don't use the wrappers that flatten the generator to a result list.

Reply:
`HostGenerator` should be the type returned.

Reply:
ack, thanks